aws/resource_aws_autoscaling_group_test.go

```diff
@@ -744,6 +745,40 @@ func TestAccAWSAutoScalingGroup_serviceLinkedRoleARN(t *testing.T) {
 	})
 }
 
+func TestAccAWSAutoScalingGroup_MaxInstanceLifetime(t *testing.T) {
+	var group autoscaling.Group
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSAutoScalingGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSAutoScalingGroupConfig_withMaxInstanceLifetime,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSAutoScalingGroupExists("aws_autoscaling_group.bar", &group),
+					resource.TestCheckResourceAttrSet(
+						"aws_autoscaling_group.bar", "max_instance_lifetime"),
```

Comment:
This should use `TestCheckResourceAttr()` to verify the actual value in the Terraform state, rather than if it was set to any (potentially erroneous) value, e.g.

```go
resource.TestCheckResourceAttr("aws_autoscaling_group.bar", "max_instance_lifetime", "864000"),
```

Reply:
Done!